SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/AbstractLockingPolicy.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "GameFramework/Actor.h"
+
+#include "SpatialConstants.h"
+
+#include "AbstractLockingPolicy.generated.h"
+
+UCLASS(abstract)
+class SPATIALGDK_API UAbstractLockingPolicy : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual bool CanAcquireLock(const AActor* Actor) const PURE_VIRTUAL(UAbstractLockingPolicy::CanAcquireLock, return false;);
```

Comment:
Do we need to expose CanAcquireLock in the interface? I can't think of many places where the user wouldn't just try to lock and then check if the return is a valid token.

Reply:
You might want this if you have a locking policy that only gives out a limited number of tokens for example.

Reply:
I guess the question is when would you call CanAquireLock, get a true return, and then not immediately afterwards call AcquireLock?

Reply:
> I guess the question is when would you call CanAquireLock, get a true return, and then not immediately afterwards call AcquireLock?

^ this makes sense to me.

In tandem with Mike's suggestion above about moving the RefCounted.cpp implementation to the parent class instead. We could have `CanAcquireLock` live in the abstract class as a virtual protected function so other policies could extend it internally but it's not part of the public interface??